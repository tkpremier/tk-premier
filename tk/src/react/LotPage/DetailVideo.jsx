import React, { Component } from 'react';
import PropTypes from 'prop-types';

class DetailVideo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      canPlay: false,
      playing: false,
      showControls: false,
      paused: false
    };
    this.checkAutoPlay = this.checkAutoPlay.bind(this);
    this.checkPlaying = this.checkPlaying.bind(this);
    this.checkPause = this.checkPause.bind(this);
  }
  checkAutoPlay(e) {
    let showControls = false;
    if (!this.state.playing) {
      e.target.play();
      showControls = true;
    }
    this.setState((state) => {
      return {
        ...state,
        showControls,
        canPlay: true
      };
    });
  }
  checkPlaying() {
    this.setState((state) => {
      return { ...state, showControls: false, playing: true };
    });
  }
  checkPause() {
    this.setState((state) => {
      return { ...state, showControls: true, paused: true, playing: false };
    });
  }
  render() {
    return (
      <video
        autoPlay
        loop
        preload="auto"
        src={this.props.src}
        onCanPlay={this.checkAutoPlay}
        controls={this.state.showControls || this.state.paused}
        onPlaying={this.checkPlaying}
        onPause={this.checkPause}
      />
    );
  }
}


DetailVideo.propTypes = {
  src: PropTypes.string
};

export default DetailVideo;