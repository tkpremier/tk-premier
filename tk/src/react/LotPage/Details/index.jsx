import React from 'react';
import classNames from 'classnames';
import isNull from 'lodash/isNull';
import LotDetailItem, { LotEssay } from './DetailItem';
import { JewelsDetails, WatchDetails } from './JewelsWatches';
import ArtistBio from './Maker';
import StickyDetails from './StickyDetails';
import { additionalContentLotPropTypes } from '../proptypes';
import ConditionReportContainer from '../../ConditionReport/ConditionReport.container';
import PhillipsTranslations from '../../PhillipsTranslations/PhillipsTranslations';
import setLineBreaks from '../../utils/setLineBreaks';
import { isTimedAuctionLive } from '../../../utils/auctionLiveState';
import PhillipsVideo from '../../PhillipsVideo/PhillipsVideo';

const enableConditionReport = ({
  auctionTimeState,
  endDate,
  isNoLot,
  saleTypeId,
  startDate,
  timeZone
}) => !isNoLot
    ? auctionTimeState === 1
    || auctionTimeState === 2
    || (saleTypeId === 3 && isTimedAuctionLive({ endDate, startDate, timeZone }))
    : false;

const LotPageDetails = (props) => {
  return props.additionalContent.length > 0
    ? (
      <StickyDetails
        {...props}
        enableConditionReport={enableConditionReport(props)}
      />
    )
    : (
      <div className="lot-page__details">
        <ul className="lot-page__details__list">
          {enableConditionReport(props)
            ? (<ConditionReportContainer {...props} key="conditionReport" />)
            : null}
          {!isNull(props.webDescription) && props.webDescription.length > 0
            ? (
              <LotDetailItem
                className="web-description"
                header="Description"
                key="web-description"
                value={props.webDescription}
              />
            )
            : null
          }
          {props.videoSource.length > 0 && !props.videoAboveTheFold
            ? (
              <li className="lot-page__details__list__item">
                <h4 className="lot-page__details__list__item__header">Video</h4>
                <PhillipsVideo
                  className={classNames({ 'phillips-video--square-format': props.videoSquareAspectRatio })}
                  description={props.videoDescription}
                  source={props.videoSource}
                  showCaption
                  title={props.videoTitle}
                />
              </li>
            )
            : null
          }
          {props.department === 'Jewelry' ? <JewelsDetails {...props} key="jewelsDetails" /> : null}
          {props.department === 'Watches' || props.isWatch ? <WatchDetails {...props} key="watchesDetails" /> : null}
          {props.provenance
            ? (
              <LotDetailItem
                header="Provenance"
                value={props.getTranslatedString('provenance', props.currentLanguage)}
                key="provenance"
              />
            )
            : null
          }
          {props.exhibited
            ? (
              <LotDetailItem
                header="Exhibited"
                value={props.getTranslatedString('exhibited', props.currentLanguage)}
                key="exhibited"
              />
            )
            : null
          }
          {props.literature && props.department !== 'Watches' && !props.isWatch
            ? (
              <LotDetailItem
                header="Literature"
                value={props.getTranslatedString('literature', props.currentLanguage)}
                key="literature"
              />

            )
            : null
          }
          {props.essay || props.cEssay
            ? <LotEssay essay={props.getTranslatedString('essay', props.currentLanguage)} key="essay" />
            : null
          }
          {(!isNull(props.artistBiography) && props.artistBiography.length > 0)
            ? (
              <ArtistBio
                key="makerBio"
                name="makerBio"
                makerNameTranslated={props.getTranslatedString('makerName', props.currentLanguage)}
                makerName={props.makerName}
                makerId={props.makerId}
                artistBiography={props.artistBiography}
                artistBirthYear={props.artistBirthYear}
                artistDeathYear={props.artistDeathYear}
                artistNationality={props.artistNationality}
                department={props.department}
              />
            )
            : null
          }
        </ul>
      </div>
    );
};
LotPageDetails.defaultProps = {
  additionalContent: []
};

LotPageDetails.propTypes = additionalContentLotPropTypes;

export default PhillipsTranslations(LotPageDetails);
