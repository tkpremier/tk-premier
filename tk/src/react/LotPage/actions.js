
const changeLot = (payload) => {
  return {
    type: 'CHANGE_LOT',
    payload
  };
};

const changeLotNoMaker = (payload) => {
  return {
    type: 'CHANGE_LOT_NO_MAKER',
    payload
  };
};

export {
  changeLot,
  changeLotNoMaker
};
