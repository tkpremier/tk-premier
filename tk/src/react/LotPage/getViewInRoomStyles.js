import { find, sample } from 'lodash';

const PERSPECTIVE_CONSTANT = 0.02;
const RACKSPACE_URL = 'https://8675fa3b12fbaae97df1-49af760ec3cce3b92480c70a3569b570.ssl.cf2.rackcdn.com/view_in_room/';
const WALL_PERCENTAGE = 0.75;
const PERSPECTIVE_DATA = [
  {
    position: 'left',
    referenceImg: `${RACKSPACE_URL}Ross_left.png`,
    heightCm: 183,
    heightPx: 549,
    widthPx: 148
  },
  {
    position: 'right',
    referenceImg: `${RACKSPACE_URL}Ross_right.png`,
    heightCm: 183,
    heightPx: 549,
    widthPx: 190
  },
  {
    position: 'left',
    referenceImg: `${RACKSPACE_URL}Jed_left.png`,
    heightCm: 185,
    heightPx: 555,
    widthPx: 273
  },
  {
    position: 'right',
    referenceImg: `${RACKSPACE_URL}Jed_right.png`,
    heightCm: 185,
    heightPx: 555,
    widthPx: 273
  },
  {
    position: 'left',
    referenceImg: `${RACKSPACE_URL}Girls_left.png`,
    heightCm: 172,
    heightPx: 516,
    widthPx: 328
  },
  {
    position: 'right',
    referenceImg: `${RACKSPACE_URL}Girls_right.png`,
    heightCm: 172,
    heightPx: 516,
    widthPx: 328
  }
];

const calcAspectRatio = (width, height) => height / width;

const getImageHeight = (lot, reference) => (
  Math.ceil(lot.height * (reference.heightPx / reference.heightCm))
);

const getImageWidth = (height, aspectRatio) => Math.ceil(height / aspectRatio);

const applyPerspective = (dimension) => Math.ceil(dimension - (dimension * PERSPECTIVE_CONSTANT));

const calcImageDimensions = (lot, reference) => {
  const height = applyPerspective(getImageHeight(lot, reference));
  return { height, width: getImageWidth(height, calcAspectRatio(lot.width, lot.height)) };
};

const getImgUrl = (lot, dimensions) => (
  { lotImgUrl: `${lot.image}/${dimensions.width}/${dimensions.height}/false/false/false` }
);

const getImgPosition = ({ height, width }) => {
  return {
    bottom: Math.ceil((window.innerHeight * WALL_PERCENTAGE - height) * 0.425),
    marginLeft: Math.ceil(width / 2)
  };
};

const calcLotImgAttributes = (lot, dimensions, wallDimensions) => {
  return {
    ...dimensions,
    ...getImgUrl(lot, dimensions),
    ...getImgPosition(dimensions, wallDimensions)
  };
};

const calcReferenceStyles = (reference, lotImgDimensions) => {
  const referenceOffset = (lotImgDimensions.width / 2 + reference.widthPx + 25);
  return { referenceOffset, ...reference };
};

const getScreenSize = () => {
  const mediaQueries = [
    { size: 'small', queryString: 'screen and (max-height: 600px)' },
    { size: 'medium', queryString: 'screen and (max-height: 850px) and (min-height: 600px)' },
    { size: 'large', queryString: 'screen and (max-height: 1280px) and (min-height: 850px)' },
    { size: 'extraLarge', queryString: 'screen and (min-height: 1280px)' }
  ];
  return find(mediaQueries, (query) => window.matchMedia(query.queryString).matches);
};

const scaleImageByViewport = ({ size }) => {
  const scales = {
    small: 0.5,
    medium: 0.75,
    large: 1,
    extraLarge: 1.25
  };
  return scales[size] || 1;
};

const scaleImageByLotHeight = ({ height }) => {
  const viewport = getScreenSize();
  const maxHeights = {
    small: 152,
    medium: 183,
    large: 213,
    extraLarge: 244
  };
  let scale = 1;
  if (height > maxHeights[viewport.size]) {
    scale = maxHeights[viewport.size] / height;
  }
  return scale;
};

const applyViewportScale = (reference) => {
  const { heightPx, widthPx } = reference;
  const scale = scaleImageByViewport(getScreenSize());
  return {
    ...reference,
    heightPx: heightPx * scale,
    widthPx: widthPx * scale
  };
};

const applyLotScale = (reference, lot) => {
  const [refHeightPx, refWidthPx] = [reference.heightPx, reference.widthPx];
  const scale = scaleImageByLotHeight(lot);
  return {
    ...reference,
    heightPx: refHeightPx * scale,
    widthPx: refWidthPx * scale
  };
};

export default (lot, reference = sample(PERSPECTIVE_DATA)) => {
  const scaledRef = applyLotScale(applyViewportScale(reference), lot);
  const lotImgDimensions = calcImageDimensions(lot, scaledRef);
  return {
    ...calcReferenceStyles(scaledRef, lotImgDimensions),
    ...calcLotImgAttributes(lot, lotImgDimensions, scaledRef)
  };
};
