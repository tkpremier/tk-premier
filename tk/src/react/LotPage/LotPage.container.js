import { connect } from 'react-redux';
import LotPage from './LotPage';
import { getAuctionMobilityLotRowIds } from './LotPageBidButton.container';
import { currentLotSelector } from './selectors';

const mapStateToProps = (state, { isServer, loginRequired }) => {
  const { modal, auction, lotNumberFull, lots, currentLanguage, bidButton, user } = state;
  return {
    auction,
    auctionMobilityLotRowIds: getAuctionMobilityLotRowIds({ lotNumberFull, lots }),
    bidButton,
    currentLanguage,
    currentLot: currentLotSelector({ lotNumberFull, lots }),
    isServer,
    loginRequired,
    lots,
    modal,
    user
  };
};

export default connect(mapStateToProps)(LotPage);