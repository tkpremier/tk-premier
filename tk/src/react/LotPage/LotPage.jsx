import React, { Component, Fragment } from 'react';
import { findIndex, trim, isUndefined, isNull, find, has, startsWith } from 'lodash/fp';
import { StickyContainer, Sticky } from 'react-sticky';
import classNames from 'classnames';
import { changeLot, changeLotNoMaker } from './actions';
import LotPageAside from './LotPageAside';
import LotPageDetails from './Details';
import LotPageImage from './LotPageImage';
import LotPageMeta from './LotPageMeta';
import LotPageNavigation from './LotPageNavigation';
import ViewInRoom from './ViewInRoom';
import { widgetConnect } from '../BidButtons/actions';
import InquireForm from '../BidButtons/Inquire/Form';
import FavoriteLot from '../FavoriteLot/FavoriteLot';
import LanguageToggle from '../LanguageToggle/LanguageToggle';
import PhillipsLotList from '../LotList/LotList.container';
import PhillipsGrid from '../PhillipsGrid/PhillipsGrid';
import PhillipsImage from '../PhillipsImage/PhillipsImage';
import PhillipsLot from '../PhillipsLot/PhillipsLot';
import PhillipsModal from '../PhillipsModal/PhillipsModal';
import OfferModalContainer from '../PhillipsModal/OfferModalContainer';
import { loggedIn } from '../PhillipsUser/actions';
import LoginPaywall from '../PhillipsUser/Forms';
import bindUserModel from '../PhillipsUser/bindUserModel';
import Share from '../Share/Share';
import SaleBanner from '../SaleBanner/SaleBanner';
import getPhillipsBackboneProperty from '../../utils/getPhillipsBackboneProperty';
import handleMql from '../../utils/handlemql';
import getAmWidget from '../utils/getAmWidget';
import lotPagePropTypes from './proptypes';

const showLanguageToggle = ({ departmentName, departments, locationName, isChineseTranslate }) => {
  return isChineseTranslate
    || (
      locationName === 'Hong Kong'
      && (
        departmentName === 'Contemporary'
        || departments.filter(dept => dept.departmentID === 1)
      )
    );
};

class LotPage extends Component {

  bbUserModel = null

  state = {
    showViewInRoom: false,
    isMobile: false
  }

  componentDidMount() {
    const { dispatch, user } = this.props;
    getPhillipsBackboneProperty('user').then((userModel) => {
      bindUserModel(userModel, dispatch);
      if (userModel.loggedIn && !user.loggedIn) {
        dispatch(loggedIn(userModel.toJSON()));
      }
      this.bbUserModel = userModel;
    });
    getPhillipsBackboneProperty('Events').then((event) => {
      event.on('loginSuccess logoutSuccess', () => {
        if (this.props.auction.saleTypeID === 3 && this.props.auction.auctionBidPartner === 1) {
          location.reload(true);
        }
      });
    });
    if (typeof matchMedia !== 'undefined') {
      handleMql('screen and (min-width: 768px)', this.setIsMobile);
    }
    if (typeof document !== 'undefined') {
      document.addEventListener('keyup', (event) => {
        const currentIndex = findIndex((lot) => {
          return lot.lotNumberFull === this.props.currentLot.lotNumberFull;
        })(this.props.lots);
        const totalCount = this.props.lots.length;
        switch (event.which) {
          case 37: {
            const prevLot = isUndefined(this.props.lots[currentIndex - 1])
              ? this.props.lots[totalCount - 1]
              : this.props.lots[currentIndex - 1];
            const { makerName, saleNumber, lotNumberFull, isNoLot } = prevLot;
            const makerUrl = isNoLot ? 'No Lot' : makerName;

            // /${maker}/${saleNumber}/${lotNumberFull}
            // uriEncoder(maker) transform happens in ./routesMap.js
            // dispatch action to change url
            dispatch(!makerName || makerName.toUpperCase() === 'NOARTIST'
              ? changeLotNoMaker({
                saleNumber,
                lotNumberFull: trim(lotNumberFull)
              })
              : changeLot({
                makerName: makerUrl,
                saleNumber,
                lotNumberFull: trim(lotNumberFull)
              })
            )
            break;
          }
          case 39: {
            const nextLot = isUndefined(this.props.lots[currentIndex + 1])
              ? this.props.lots[0]
              : this.props.lots[currentIndex + 1];
            const { makerName, saleNumber, lotNumberFull, isNoLot } = nextLot;
            const makerUrl = isNoLot ? 'No Lot' : makerName;
            // /${maker}/${saleNumber}/${lotNumberFull}
            // uriEncoder(maker) transform happens in ./routesMap.js
            // dispatch action to change url

            dispatch(!makerName || makerName.toUpperCase() === 'NOARTIST'
              ? changeLotNoMaker({
                saleNumber,
                lotNumberFull: trim(lotNumberFull)
              })
              : changeLot({
                makerName: makerUrl,
                saleNumber,
                lotNumberFull: trim(lotNumberFull)
              })
            );
            break;
          }
          default: {
            // do nothing
          }
        }
      });
    }
  }

  componentWillReceiveProps(nextProps) {
    if (trim(nextProps.currentLot.lotNumberFull) !== trim(this.props.currentLot.lotNumberFull)) {
      window.scroll(0, 0);
      this.setState(state => ({
        isMobile: state.isMobile,
        showViewInRoom: false
      }));
    }
  }

  componentDidUpdate(prevProps, prevState) {
    // check to see if it's Timed Auction with AM
    if (this.props.auction.saleTypeID === 3 && this.props.auction.auctionBidPartner === 1) {
      // check to see if isMobile or if it's desktop and re-initializing
      if (
        (!has('detailLotWidget')(window) && (this.state.isMobile || prevState.isMobile === this.state.isMobile))
        || (prevProps.currentLot.auctionMobilityLotRowId !== this.props.currentLot.auctionMobilityLotRowId)
        || (startsWith('widgetConnect')(this.props.bidButton.status)
          && prevProps.bidButton.status !== this.props.bidButton.status
          && !(prevProps.bidButton.status === '' && this.props.bidButton.status === 'widgetConnectSuccess')
        )
      ) {
        this.setAmWidget();
      }
    }
    return !isNull(this.bbUserModel)
      ? (this.props.user.loggedIn && !this.bbUserModel.loggedIn)
        ? this.bbUserModel.trigger('reduxUpdate', this.props.user)
        : null
      : null;
  }

  setAmWidget() {
    const { dispatch } = this.props;
    if (!has('detailLotWidget')(window)) {
      getAmWidget()
        .then(({ AmWidget }) => {
          window.detailLotWidget = new AmWidget({
            auctionId: this.props.currentLot.auctionMobilityAuctionRowId,
            url: window.localStorage.getItem('amApiUrl'),
            wrapperPrefix: 'detail-am-target',
            layout: 'detail',
            lots: this.props.auctionMobilityLotRowIds
          });
          window.detailLotWidget.init();
          window.detailLotWidget.setWidgetHandler('system', ({ websocket_state }) => {
            switch (websocket_state) {
              case 'errored':
                dispatch(widgetConnect('ERROR'));
                break;
              case 'disconnected':
                dispatch(widgetConnect('FAIL'));
                break;
              case 'connected':
                dispatch(widgetConnect('SUCCESS'));
                break;
              default:
                break;
            }
          });
        })
        .catch((error) => {
          console.log('AM Error: ', error);
        });
    } else {
      window.detailLotWidget.loadLots([this.props.currentLot.auctionMobilityLotRowId]);
    }
  }

  setIsMobile = (mql) => {
    if (mql.matches) {
      if (this.state.isMobile) {
        this.setState(state => ({ ...state, isMobile: false }));
      } else {
        if (this.props.auction.saleTypeID === 3 && this.props.auction.auctionBidPartner === 1) {
          this.setAmWidget();
        }
      }
      return false;
    }
    if (!this.state.isMobile) {
      this.setState(state => ({ ...state, isMobile: true }));
    }
    return this.state.isMobile;
  }

  getModal = (type) => {
    switch (type) {
      case 'cloudinary':
        return (
          <PhillipsModal
            {...this.props.modal}
            customClasses={['phillips-modal--image']}
          >
            <PhillipsImage
              alt={this.props.modal.alt}
              cloudinary
              imagePath={this.props.modal.imagePath}
              transformation="TwoRows"
            />
          </PhillipsModal>
        )
      case 'html':
        return (
          <PhillipsModal
            {...this.props.modal}
            customClasses={['phillips-modal--html']}
          >
            <div dangerouslySetInnerHTML={{ __html: this.props.modal.html }} />
          </PhillipsModal>
        );
      case 'offer':
        return (
          <PhillipsModal
            {...this.props.modal}
            customClasses={['offer-modal']}
          >
            <OfferModalContainer />
          </PhillipsModal>
        );
      case 'iframe':
        return (
          <PhillipsModal
            {...this.props.modal}
            customClasses={['phillips-modal--iframe']}
          >
            <iframe
              allow="fullscreen"
              height={this.props.modal.height}
              name={this.props.modal.name}
              src={this.props.modal.src}
              title={this.props.modal.name}
              width={this.props.modal.width}
            />
          </PhillipsModal>
        );
      case 'inquire':
        return (
          <PhillipsModal
            {...this.props.modal}
            customClasses={['phillips-modal--inquire']}
          >
            <InquireForm {...this.props.currentLot} formOpen />
          </PhillipsModal>
        );

      default: return null;
    }
  }

  render() {
    const { loginRequired, user } = this.props;
    let languageToggle = null;
    let stickyBtmOffset = 125;
    const toggleViewInRoom = (e) => {
      e.preventDefault();
      this.setState(state => ({ ...state, showViewInRoom: !state.showViewInRoom }));
    };

    let auctionMobilityLotRowIds = this.props.lots.map(lot => {
      if (has('auctionMobilityLotRowId')(lot)) {
        return lot.auctionMobilityLotRowId;
      }
    });
    auctionMobilityLotRowIds = auctionMobilityLotRowIds.slice(0, 10);
    if (showLanguageToggle(this.props.auction)) {
      languageToggle = <LanguageToggle />;
      // adding language toggle height to sticky container bottom offset so it doesn't cut off contact info
      stickyBtmOffset = 125 + 66;
    }
    // swipe event handler to navigate through lots
    // const handleSwipe = (e) => {
    //   if (e.direction === 2 || e.direction === 4) {
    //     const int = e.direction === 2 ? 1 : -1;
    //     const currentIndex = findIndex((lot) => {
    //       return lot.lotNumberFull === this.props.currentLot.lotNumberFull;
    //     })(this.props.lots);
    //     const { maker, saleNumber, lotNumberFull, isNoLot } = this.props.lots[currentIndex + int];
    //     const makerUrl = (isNoLot) ? 'No Lot' : maker;
    //
    //     this.props.dispatch(changeLot({
    //       maker: makerUrl,
    //       saleNumber,
    //       lotNumberFull: trim(lotNumberFull)
    //     }));
    //   }
    // };
    const viewInRoomInfo = find(image => image.isViewInRoom === true)(
      this.props.currentLot.lotImages
    );
    return loginRequired && !user.loggedIn
      ? (
        <div className="main-container">
          <LoginPaywall user={user} />
        </div>
      )
      : (
        <div
          className={classNames('lot-page main-container', { viewInRoom: this.state.showViewInRoom })}
        >
          {/* boolean to add client-side metatag updates via react-helmet */}
          {this.props.isServer ? null : (
            <LotPageMeta {...this.props.currentLot} auction={this.props.auction} />
          )}
          {this.props.auction.saleTypeID !== 5
            ? (
              <SaleBanner
                currentLanguage={this.props.currentLanguage}
                {...this.props.auction}
                truncateTitle
              />
            )
            : null
          }
          {this.props.modal.show
            ? this.getModal(this.props.modal.type)
            : null
          }

          {this.state.showViewInRoom ? (
            <ViewInRoom
              image={viewInRoomInfo.imagePath}
              width={this.props.currentLot.width}
              height={this.props.currentLot.height}
              transformation="LotDetailMainImage"
              cloudinary={this.props.auction.useCloudinary}
              description={this.props.currentLot.description}
              onClose={toggleViewInRoom}
            />
          ) : null}
          <div className="container content-area has-sale-banner">
            <div className="col-xs-12">
              <div className="lot-detail-content row">
                <div className="left-column col-xs-12 col-md-7">
                  {this.state.isMobile && this.props.auction.saleTypeID !== 5
                    ? (
                      <div className="page-controls">
                        <LotPageNavigation
                          saleNumber={this.props.auction.saleNumber}
                          lots={this.props.lots}
                          currentLotNumberFull={this.props.currentLot.lotNumberFull}
                          dispatch={this.props.dispatch}
                        />
                        {languageToggle}
                      </div>
                    )
                    : null
                  }
                  {this.props.currentLot.isNoLot
                    ? null
                    : (
                      <LotPageImage
                        {...this.props.currentLot}
                        imagePath={this.props.currentLot.lotImages[0].imagePath}
                        isMobile={this.state.isMobile}
                      />
                    )
                  }
                  {this.props.currentLot.isNoLot === false ? (
                    <PhillipsLotList
                      key={`lot-list-main-${this.props.currentLot.lotNumberFull.trim()}-${this.props.currentLot.saleNumber}`}
                      lot={this.props.currentLot}
                    >
                      {eventHandlers => (
                        <div className="phillips-social">
                          <FavoriteLot
                            {...this.props.currentLot}
                            onPress={eventHandlers.onPress}
                            onMouseEnter={eventHandlers.onMouseEnter}
                            onMouseLeave={eventHandlers.onMouseLeave}
                          />
                          {viewInRoomInfo && !this.state.isMobile ? (
                            <div
                              role="button"
                              tabIndex={0}
                              className="view-in-room-button"
                              onClick={toggleViewInRoom}
                            >
                              <span className="icon" />
                              <span className="tooltip">View in Room</span>
                            </div>
                          ) : null}
                          <Share path={this.props.currentLot.detailLink} />
                        </div>
                      )}
                    </PhillipsLotList>
                  ) : null}
                  {this.state.isMobile ? (
                    <LotPageAside
                      {...this.props.currentLot}
                      auctionBidPartner={this.props.auction.auctionBidPartner}
                      auctionContactEmail={this.props.auction.contactEmail}
                      auctionTimeState={this.props.auction.timeState}
                      conditionRequestEmail={this.props.auction.conditionRequestEmail}
                      currentLanguage={this.props.currentLanguage}
                      endDate={this.props.auction.endDate}
                      endSale={this.props.auction.endSale}
                      isMobile={this.state.isMobile}
                      saleTypeId={this.props.auction.saleTypeID}
                      showAdvanceBidButton={this.props.currentLot.showAdvanceBidButton}
                      startDate={this.props.auction.startDate}
                    />
                  ) : null}

                  <LotPageDetails
                    {...this.props.currentLot}
                    auctionTimeState={this.props.auction.timeState}
                    conditionRequestEmail={this.props.auction.conditionRequestEmail}
                    currentLanguage={this.props.currentLanguage}
                    endDate={this.props.auction.endDate}
                    saleTypeId={this.props.auction.saleTypeID}
                    showAdvanceBidButton={this.props.currentLot.showAdvanceBidButton}
                    startDate={this.props.auction.startDate}
                    timeZone={this.props.auction.timeZone}
                  />
                </div>
                {this.state.isMobile
                  ? null
                  : (
                    <div
                      className={classNames('right-column col-xs-12 col-md-5', {
                        'widget-enabled':
                          this.props.auction.saleTypeID === 3 &&
                          this.props.auction.auctionBidPartner === 1
                      })}
                    >
                      <div className="page-controls">
                        {this.props.auction.saleTypeID !== 5
                          ? (
                            <LotPageNavigation
                              saleNumber={this.props.saleNumber}
                              lots={this.props.lots}
                              currentLotNumberFull={this.props.currentLot.lotNumberFull}
                              dispatch={this.props.dispatch}
                            />
                          )
                          : null
                        }
                        {languageToggle}
                      </div>
                      <StickyContainer className="sticky-container">
                        <Sticky
                          topOffset={-95}
                          bottomOffset={stickyBtmOffset}
                        >
                          {(props) => {
                            let styles = props.style;
                            if (props.isSticky) {
                              styles = { ...styles, top: styles.top + 95 };
                            }
                            return (
                              <div className="sticky-container__sticky-wrapper" style={styles}>
                                <LotPageAside
                                  {...this.props.currentLot}
                                  currentLanguage={this.props.currentLanguage}
                                  auctionTimeState={this.props.auction.timeState}
                                  showAdvanceBidButton={this.props.currentLot.showAdvanceBidButton}
                                  saleTypeId={this.props.auction.saleTypeID}
                                  startDate={this.props.auction.startDate}
                                  endDate={this.props.auction.endDate}
                                  endSale={this.props.auction.endSale}
                                  auctionBidPartner={this.props.auction.auctionBidPartner}
                                  conditionRequestEmail={this.props.auction.conditionRequestEmail}
                                  isMobile={this.state.isMobile}
                                />
                              </div>
                            );
                          }}
                        </Sticky>
                      </StickyContainer>
                    </div>
                  )
                }
              </div>
            </div>
            {this.props.auction.saleTypeID !== 5
              ? (
                <Fragment>
                  <div className="auction-grid-header col-xs-12">
                    <h2 dangerouslySetInnerHTML={{ __html: this.props.auction.auctionTitle }} />
                    <p dangerouslySetInnerHTML={{ __html: this.props.auction.auctionDetailsSmall }} />
                  </div>
                  <div className="auction-grid col-xs-12">
                    <PhillipsGrid columns={{ lg: 3, md: 4, sm: 6, xs: 6 }}>
                      {this.props.lots.map(lotData => (
                        <PhillipsLot
                          key={`grid-${lotData.lotNumberFull.trim()}-${lotData.saleNumber}`}
                          currentLanguage={this.props.currentLanguage}
                          useCloudinary={this.props.auction.useCloudinary}
                          hasRouter
                          lazyLoadOffset={0}
                          {...lotData}
                          auctionMobilityLotRowIds={auctionMobilityLotRowIds}
                          saleTypeId={this.props.auction.saleTypeID}
                        />
                      ))}
                    </PhillipsGrid>
                  </div>
                </Fragment>
              )
              : null
            }
          </div>
        </div>
      );
  }
}

LotPage.defaultProps = {
  auctionMobilityLotRowIds: [],
  currentLanguage: 'en-US',
  modal: {
    payload: {},
    show: false,
    type: ''
  }
};

LotPage.propTypes = lotPagePropTypes;

export default LotPage;
