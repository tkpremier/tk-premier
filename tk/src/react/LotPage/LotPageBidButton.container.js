import { createSelector } from 'reselect';
import { connect } from 'react-redux';
import { findIndex, isUndefined } from 'lodash/fp';
import BidButton from '../BidButtons/BidButton';
import { showModal } from '../PhillipsModal/actions';
import { submitOffer, offerReject } from '../BidButtons/actions';

export const lotNumberFullSelector = state => state.lotNumberFull;

export const lotsSelector = state => state.lots;

const getPrevLotIds = (currentIndex, lots, count) => {
  const prevIds = [];
  const totalLotsCount = lots.length;
  const maxCount = totalLotsCount > count
    ? count
    : totalLotsCount;
  for (let i = 0; i < maxCount; i++) {
    const lot = isUndefined(lots[currentIndex - (i + 1)]) ?
      lots[totalLotsCount - (i + 1)] :
      lots[currentIndex - (i + 1)];
    prevIds.push(lot.auctionMobilityLotRowId);
  }
  return prevIds;
};

const getNextLotIds = (currentIndex, lots, count) => {
  const nextIds = [];
  const totalLotsCount = lots.length;
  const maxCount = totalLotsCount > count
    ? count
    : totalLotsCount;
  for (let i = 0; i < maxCount; i++) {
    const lot = isUndefined(lots[currentIndex + (i + 1)]) ?
      lots[totalLotsCount - (i + 1)] :
      lots[currentIndex + (i + 1)];
    nextIds.push(lot.auctionMobilityLotRowId);
  }
  return nextIds;
};

export const getAuctionMobilityLotRowIds = createSelector(
  [lotNumberFullSelector, lotsSelector],
  (lotNumberFull, lots) => {
    const currentIndex = findIndex(lot => lot.lotNumberFull.trim() === lotNumberFull.trim())(lots);
    const currentLotRowId = lots[currentIndex].auctionMobilityLotRowId;
    const prevLotIds = getPrevLotIds(currentIndex, lots, 20);
    prevLotIds.push(currentLotRowId);
    return prevLotIds.concat(getNextLotIds(currentIndex, lots, 20));
  });

export const getLotRowIds = createSelector(
  [lotsSelector],
  lots => lots.map(lot => lot.auctionMobilityLotRowId)
);

const mapStateToProps = ({ auction, user, bidButton, lots, lotNumberFull }, ownProps) => {
  const { contactEmail } = auction;
  return {
    contactEmail,
    ...bidButton,
    ...ownProps,
    user,
    auctionMobilityLotRowIds: getAuctionMobilityLotRowIds({ lotNumberFull, lots })
  };
};

const mapDispatchToProps = {
  showModal,
  submitOffer,
  offerReject
};

export default connect(mapStateToProps, mapDispatchToProps)(BidButton);
