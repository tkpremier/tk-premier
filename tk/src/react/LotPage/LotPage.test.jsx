import React from 'react';
import LotPage from './LotPage';
import renderer from 'react-test-renderer';

describe('LotPageContainer', () => {
  it('should render a LotPage', () => {
    const lotPageContainer = renderer.create(
      <LotPage />
    );
    const tree = lotPageContainer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});