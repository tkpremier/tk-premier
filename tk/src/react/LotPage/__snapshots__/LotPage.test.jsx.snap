// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LotPageContainer should render a LotPage 1`] = `
<div
  className="lot-page main-container"
>
  <div
    className="container content-area"
  >
    <div
      className="left-column col-xs-12 col-md-7"
    >
      <h1>
        LotPageImage
      </h1>
      <h1>
        PhillipsSocial
      </h1>
      <h1>
        LotPageAside
      </h1>
      <h1>
        LotPageDetail
      </h1>
    </div>
    <div
      className="right-column col-xs-12 col-md-5"
    >
      <h1>
        LanguageToggle
      </h1>
      <h1>
        Pagination
      </h1>
      <h1>
        StickyContainer
      </h1>
    </div>
    <div
      className="auction-grid col-xs-12"
    >
      <h1>
        Phillips Grid
      </h1>
    </div>
  </div>
</div>
`;
